### ISSUE TITLE
activitity overview + screen lock crashed gnome shell (fedora 40, fedora 41) when dash to dock extension is enabled

### ISSUE THREAD

**Issue by kk0nrad** [micheleg/dash-to-dock#2338]
It's a very strange behaviour that I noticed when on fedora 40 and that still persists with fedora 41: I usually lock the screen with windows key + L, and sometimes it happens that I press windows key then windows key + L by mistake. When this happens, gnome shell crashes. I was able to track down the correlation with dash to dock because when the extension is disabled this does not happen at all.

**Comment by kk0nrad** [comment]
btw: using wayland

**Comment by zak905** [comment]
I am experiencing the same. I checked the gnome shell logs, and I found the following:
```
Nov 30 21:15:12 fedora gnome-shell[10580]: Connection to xwayland lost
Nov 30 21:15:12 fedora gnome-shell[10580]: Xwayland terminated, exiting since it was mandatory
Nov 30 21:15:12 fedora gnome-shell[10580]: JS ERROR: Gio.IOErrorEnum: Xwayland exited unexpectedly
   @resource:///org/gnome/shell/ui/init.js:21:20
```

**Comment by mrme8895** [comment]
Does this only happen on Fedora?

**Comment by sergio-costas** [comment]
I think that this was a bug in Gnome Shell itself... At least, I think that this same bug was commented here some months ago.

**Comment by sergio-costas** [comment]
Yeah, this was the PR that I did as a workaround... in the comments it is explained the bug itself and the fix in gnome shell https://github.com/micheleg/dash-to-dock/pull/2219

**Comment by kk0nrad** [comment]
> Yeah, this was the PR that I did as a workaround... in the comments it is explained the bug itself and the fix in gnome shell #2219

how can the fix be implemented? is there something I can modify on the fly? If the issue still manifests itself it means that the main bug hasn't been fixed yet, so I guess we sill need the modified behaviour for leaving overview mode before locking the screen.
